standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -7109,22 +7121,37 @@ protected String describeResult() {
   }
 
   @Override
-  public List<HiveObjectPrivilege> listPrincipalDBGrantsAll(
-      String principalName, PrincipalType principalType) {
-    try (QueryWrapper queryWrapper = new QueryWrapper()) {
-      return convertDB(listPrincipalAllDBGrant(principalName, principalType, queryWrapper));
+  public List<HiveObjectPrivilege> listPrincipalDBGrantsAll(String principalName, PrincipalType principalType) {
+    List<HiveObjectPrivilege> results = Collections.emptyList();
+    try {
+      openTransaction();
+      results = convertDB(listPrincipalAllDBGrant(principalName, principalType));
+      commitTransaction();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    } finally {
+      rollbackAndCleanup(true, null);
     }
+    return results;
   }
 
   @Override
   public List<HiveObjectPrivilege> listDBGrantsAll(String catName, String dbName) {
-    return listDBGrantsAll(catName, dbName, null);
+    List<HiveObjectPrivilege> results = Collections.emptyList();
+    try {
+      openTransaction();
+      results = listDBGrantsAll(catName, dbName, null);
+      commitTransaction();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    } finally {
+      rollbackAndCleanup(true, null);
```

Comment:
Why pass true to rollbackAndCleanup here?

Reply:
I'm sorry?

Reply:
Sorry for the delay, my concern is that set success to true on ```rollbackAndCleanup``` here may make the transaction unable to rollback.

Reply:
OK.  I just took a look at it.  'true' here means 'do not roll back'.  This particular method is a top-level method and it is read-only.  Therefore, there will never be anything to roll back.